src/plugins/intel_cpu/src/nodes/qkv_proj.cpp

```diff
@@ -123,145 +157,245 @@ struct QKVProjection::Impl {
 
+            // make sure scrach is big enough
+            auto newMemDesc = std::make_shared<CpuBlockedMemoryDesc>(ov::element::u8, Shape{allocator.size()});
+            m_scratchMem = m_scrachPad->createScratchPadMem(newMemDesc);
             m_scratch_base = m_scratchMem->getDataAs<uint8_t>();
-            for (size_t ithr = 0; ithr < works.size(); ithr++) {
-                auto& work = works[ithr];
-                if (work) {
-                    work.set_C(M, reinterpret_cast<float*>(m_scratch_base + scratch_offsets[ithr]));
-                }
-            }
 
+            allocator.finalize(m_scratch_base);
             m_M = M;
         }
     }
 
-    void execute() {
-        static ReduceAdd2bh jit_2bh(false);
+    void execute() override {
+        static ReduceAdd2bh jit_cvt(false, std::is_same<T, ov::float16>::value);
+
         auto input = m_node->getSrcMemoryAtPort(0);
         const auto& ishape = input->getStaticDims();
-        uint8_t* pA = input->getDataAs<uint8_t>();
+        uint8_t* psrc0 = input->getDataAs<uint8_t>();
         int M = shape_size(ishape) / ishape[ishape.size() - 1];
-        auto* dst0 = m_node->getDstMemoryAtPort(0)->getDataAs<ov::bfloat16>();
-        auto* dst1 = m_node->getDstMemoryAtPort(1)->getDataAs<ov::bfloat16>();
-        auto* dst2 = m_node->getDstMemoryAtPort(2)->getDataAs<ov::bfloat16>();
+        auto* dst0 = m_node->getDstMemoryAtPort(0)->getDataAs<T>();
+        auto* dst1 = m_node->getDstMemoryAtPort(1)->getDataAs<T>();
+        auto* dst2 = m_node->getDstMemoryAtPort(2)->getDataAs<T>();
+
+        float* w_scale[3];
+
+        if (m_node->m_config.quantized) {
+            w_scale[0] = m_node->getSrcMemoryAtPort(4)->getDataAs<float>();
+            if (m_node->m_config.weights_combined) {
+                w_scale[1] = w_scale[0] + m_node->m_config.proj_size0;
+                w_scale[2] = w_scale[1] + m_node->m_config.proj_size1;
+            } else {
+                w_scale[1] = m_node->getSrcMemoryAtPort(5)->getDataAs<float>();
+                w_scale[2] = m_node->getSrcMemoryAtPort(6)->getDataAs<float>();
+            }
+        }
 
         const auto& srcStrides = input->getDescWithType<BlockedMemoryDesc>()->getStrides();
         const auto& dstStrides0 = m_node->getDstMemoryAtPort(0)->getDescWithType<BlockedMemoryDesc>()->getStrides();
         const auto& dstStrides1 = m_node->getDstMemoryAtPort(1)->getDescWithType<BlockedMemoryDesc>()->getStrides();
         const auto& dstStrides2 = m_node->getDstMemoryAtPort(2)->getDescWithType<BlockedMemoryDesc>()->getStrides();
 
-        int strideA = srcStrides[1] * sizeof(ov::bfloat16);
-        auto stride_0 = dstStrides0[1];
-        auto stride_1 = dstStrides1[1];
-        auto stride_2 = dstStrides2[1];
+        int stride_src = srcStrides[1] * sizeof(T);
+        auto stride_dst_0 = dstStrides0[1];
+        auto stride_dst_1 = dstStrides1[1];
+        auto stride_dst_2 = dstStrides2[1];
 
+        auto asym = true;
```

Comment:
Just for note: here(+quantization in MatrixDynQuantPerRow) force the activation asym quantized.